### PR TITLE
Add esr and esr_previous to api/lando/uplift/train

### PR DESCRIPTION
**Pull request by DonalMe**
I think it makes sense to use RELEASE here for the corresponding ESR versions

### CODE REVIEW

File: app/classes/ReleaseInsights/API/LandoUpliftTrain.php [hunk 1]
```diff
@@ -54,6 +56,12 @@ public function getTrains(): array
         $nightly = RELEASE + 2;
         $beta    = RELEASE + 1;
 
+        // Current ESR and the previous ESR branch (major versions).
+        // $esr_previous is null once the previous ESR reaches end-of-life.
+        $esr          = (int) Version::getMajor(ESR::getVersion(RELEASE));
+        $older_esr    = ESR::getOlderSupportedVersion(RELEASE);
+        $esr_previous = $older_esr !== null ? (int) Version::getMajor($older_esr) : null;
+
```
Comment (pascalchevrel) [inline app/classes/ReleaseInsights/API/LandoUpliftTrain.php]: We already have constants in app/inc/config.php for ESR, we can reuse them and extend as needed. I would prefer that versions definitions remain centralized.

File: app/classes/ReleaseInsights/API/LandoUpliftTrain.php [hunk 2]
```diff
@@ -96,6 +104,10 @@ public function getTrains(): array
                 'version'      => RELEASE,
                 'release_date' => $date(FIREFOX_RELEASE),
             ],
+            'esr' => [
+                'version' => $esr,
+            ],
+            'esr_previous' => $esr_previous !== null ? ['version' => $esr_previous] : null,
```
Comment (pascalchevrel) [inline app/classes/ReleaseInsights/API/LandoUpliftTrain.php]: This is changing the json schema on the existence of the variable. The 'version' field should always be available to the consumer, only the value it stores should change.

File: tests/Unit/ReleaseInsights/API/LandoUpliftTrainTest.php [hunk 3]
```diff
@@ -48,4 +48,17 @@
...
+    expect($obj->getTrains()['esr'])
+        ->toHaveCount(1)
+        ->toHavekeys(['version']);
+    expect($obj->getTrains()['esr']['version'])
+        ->toBeInt();
+    if ($obj->getTrains()['esr_previous'] !== null) {
+        expect($obj->getTrains()['esr_previous'])
+            ->toHavekeys(['version']);
+        expect($obj->getTrains()['esr_previous']['version'])
+            ->toBeInt();
```
Comment (pascalchevrel) [inline tests/Unit/ReleaseInsights/API/LandoUpliftTrainTest.php]: This test will fail when we have only one ESR, for the reason that I mentioned earlier about the missing 'version' field if there is $esr_previous is empty, but also because in the class we return null in this case, the test should test for int or null.
